### DISCUSSION TITLE
tutorials/scripting/gdextension/gdextension_docs_system

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#169] [User-contributed notes]
# tutorials/scripting/gdextension/gdextension_docs_system

The GDExtension documentation system works in a similar manner to the built-in engine documentation. It uses a series of XML files (one per class) to document the exposed constructors, properties, ...

https://docs.godotengine.org/en/stable/tutorials/scripting/gdextension/gdextension_docs_system.html

**Comment by EdwardChanCH** [comment]
Adding those extra lines to the SConstruct file would cause errors, when compiled with "scons platform=windows target=template_debug debug_symbols=yes": 
```
(...normal SCons outputs up to here)
Linking Shared Library game\bin\libgdexample.windows.template_debug.x86_64.dll ...
src\gen\doc_data.gen.windows.template_debug.x86_64.obj : warning LNK4042: object specified more than once; extras ignored
LINK : error LNK1218: warning treated as error; no output file generated
scons: *** [game\bin\libgdexample.windows.template_debug.x86_64.dll] Error 4042
scons: building terminated because of errors.
```

**Reply by EdwardChanCH** [reply]
Update:
I found the cause!
My recursive search added "src/gen/doc_data.gen.cpp" to the sources list, which crashed SCons.
Excluding "src/gen/*" from the recursive search fixed it.
[SConstruct_fixed.txt](https://github.com/user-attachments/files/17673465/SConstruct_fixed.txt)

**Comment by TheOathMan** [comment]
It's important to note that the `SConstruct` code that allow the editor to load your docs should be inserted after this line:
```
sources = Glob("src/*.cpp")
```
I lost good time trying to figure out why my docs isn't loading, turns out, the code shouldn't be appended at the end.

**Reply by EdwardChanCH** [reply]
Hi! 
The line you mentioned doesn't exist in my SConstruct, since it has been modified to support recursive searching of .cpp files in "src/". I tried inserting the code after where that line originally was but to no avail. 
[SConstruct_code.txt](https://github.com/user-attachments/files/17603852/SConstruct_code.txt)

**Reply by TheOathMan** [reply]
they have updated the [SConsttruct](https://docs.godotengine.org/en/stable/_downloads/45a3f5e351266601b5e7663dc077fe12/SConstruct) file in the 4.3 [Example](https://docs.godotengine.org/en/stable/tutorials/scripting/gdextension/gdextension_cpp_example.html) page of.

**Reply by EdwardChanCH** [reply]
I'm using the new SConstruct file, but it still shows errors after adding the recursive search to it.

**Reply by em9836** [reply]
does anyone know how to setup VSCode for this to work?
I keep getting errors in the code.

**Comment by coljac** [comment]
Running `Godot_v4.4-stable_linux.x86_64 --doctool ../ --gdexension-doc` I got `doc/classes/*xml`, not `doc_classes/`. The 4.3 docs for this page suggest `docs/classes/`, so this is a bit confusing.

**Reply by Calinou** [reply]
`doc_classes` is what's used in each module's folder (e.g. [`modules/text_server_adv/doc_classes`](https://github.com/godotengine/godot/tree/master/modules/text_server_adv/doc_classes)). The core classes are in `doc/classes` instead.

(Note that the `text_server_adv` module I linked can also be compiled as a GDExtension.)

GDExtensions typically use `doc_classes` because their structure follows those of [C++ modules](https://docs.godotengine.org/en/stable/contributing/development/core_and_modules/custom_modules_in_cpp.html).
